src/journal/views.py

```diff
@@ -1082,13 +1089,29 @@ def publish_article(request, article_id):
                 modal = 'pubdate'
 
         if 'author' in request.POST:
-            logic.notify_author(request, article)
-            return redirect(
-                reverse(
-                    'publish_article',
-                    kwargs={'article_id': article.pk},
-                )
+            notify_author_form = forms.PrepubNotifyAuthorForm(
+                request.POST,
+                request.FILES,
+                setting_name='author_publication',
+                email_context={
+                    'article': article
+                },
+                request=request,
             )
+            if notify_author_form.is_valid():
+                logic.notify_author(request, article)
```

Comment:
This function has not been updated to handle the `SettingEmailForm` argument. it is instead reading off the body and handing it to the even framework.

As a result, the `CC` and `subject` parts of the form will be ignored, which might lead to frustration from the editor.

In order to resolve this `notify_author` and the events listening to `ON_AUTHOR_PUBLICATION` need to be refactored and make use of the result from `SettingEmailForm.as_dataclass()` method

Reply:
Thanks @mauromsl -- I have gone ahead and significantly refactored this to implement the usability upgrades we need more fully. See the updated PR description.